sql/information_schema/columns_table.go

```diff
@@ -342,7 +366,21 @@ func getColumnDefault(ctx *sql.Context, cd *sql.ColumnDefaultValue) interface{}
 
 	switch l := v.(type) {
 	case time.Time:
-		v = l.Format("2006-01-02 15:04:05")
+		if sql.IsDateType(cd.Type()) {
+			v = l.Format("2006-01-02")
+		} else {
+			v = l.Format("2006-01-02 15:04:05")
+		}
```

Comment:
(minor) since we don't currently distinguish between date/timestamp/datetime, it's probably better to default to the more specific timestamp format, instead of the date format.